### ISSUE TITLE
feature: customization (as implemented in ks 0.14) is required for MCC deployment and configuration of scoped resources

### ISSUE THREAD

**Issue by clubanderson** [kubestellar/kubestellar#1886]
### Feature Description

the lack of support for customization in ks 0.2x is causing issues when supporting deployment/configuration of projects that are multi-cluster aware and require environment variables to properly connect hub/spoke components

### Proposed Solution

similar (exactly the same) as in ks 0.14, we had the ability to add annotations (in the form of k/v pairs) to inventory objects and then reference those k/v pairs in object definitions

here is how a reference works: (note the '%(remoteWriteURL)')

```
apiVersion: v1
kind: Namespace
metadata:
  name: commonstuff
---
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: commonstuff
  name: httpd-htdocs
  annotations:
    edge.kubestellar.io/expand-parameters: "true"
data:
  index.html: |
    <!DOCTYPE html>
    <html>
      <body>
        This is a common web site.
        Running in %(remoteWriteURL).
      </body>
    </html>
---

```

annotations in the inventory are in the form of k/v pairs:

```
remoteWriteURL=http://thanos-receiver-router-monitoring.kscore-stage-90.us-south.containers.appdomain.cloud/api/v1/receive"

remoteWriteURL=http://thanos-receiver-router-monitoring.kscore-prod-90.us-south.containers.appdomain.cloud/api/v1/receive"

remoteWriteURL=http://thanos-receiver-router-monitoring.mcc-kscore-stage-90.us-south.containers.appdomain.cloud/api/v1/receive"

remoteWriteURL=http://thanos-receiver-router-monitoring.mcc-kscore-prod-90.us-south.containers.appdomain.cloud/api/v1/receive"

remoteWriteURL=http://thanos-receiver-router-monitoring.aws-core-90.us-south.containers.appdomain.cloud/api/v1/receive"

remoteLokiURL=http://thanos-receiver-router-monitoring.edgeplatform-test-kscore-90.us-south.containers.appdomain.cloud/api/v1/receive"

remoteLokiURL=https://logging-loki-openshift-logging.kscore-stage-90.us-south.containers.appdomain.cloud/api/logs/v1/application"

remoteLokiURL=https://logging-loki-openshift-logging.kscore-prod-90.us-south.containers.appdomain.cloud/api/logs/v1/application"

remoteLokiURL=https://logging-loki-openshift-logging.mcc-kscore-stage-90.us-south.containers.appdomain.cloud/api/logs/v1/application"

remoteLokiURL=https://logging-loki-openshift-logging.mcc-kscore-prod-90.us-south.containers.appdomain.cloud/api/logs/v1/application"

remoteLokiURL=https://logging-loki-openshift-logging.aws-core-90.us-south.containers.appdomain.cloud/api/logs/v1/application"

remoteLokiURL=https://logging-loki-openshift-logging.edgeplatform-test-kscore-90.us-south.containers.appdomain.cloud/api/logs/v1/application"

```

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

@MikeSpreitzer please embellish to add more detail

**Comment by clubanderson** [comment]
without this support, each object using these vars needs to be stored in source control per cluster. In the environment I am managing this is 14 clusters. 7 hubs with 2 wecs per hub - and growing

**Comment by MikeSpreitzer** [comment]
My understanding of the problem @clubanderson is facing is as follows. He has seven environments, each with one WDS and two WECs. He wants to maintain just one copy of the YAML for his workload objects. There are two points where customization is needed. These can be seen in https://github.ibm.com/MCC/IRIS/blob/ae282c5cb0d19bab06688c2fa17c65ca4ea04297/log_pipeline/deployment/leaf_cluster/all-clusters/ClusterLogForwarder.yaml .

1. One is in https://github.ibm.com/MCC/IRIS/blob/ae282c5cb0d19bab06688c2fa17c65ca4ea04297/log_pipeline/deployment/leaf_cluster/all-clusters/ClusterLogForwarder.yaml#L28, where the name of the WEC needs to appear in the workload object.
2. The other customization point is in https://github.ibm.com/MCC/IRIS/blob/ae282c5cb0d19bab06688c2fa17c65ca4ea04297/log_pipeline/deployment/leaf_cluster/all-clusters/ClusterLogForwarder.yaml#L23, for customizing to the environment rather than the WEC. This puts an environment-specific URL for Loki ingestion into the workload object.

**Comment by clubanderson** [comment]
closing, no longer needed. Only took 10 lines of code to implement.

**Comment by MikeSpreitzer** [comment]
What @clubanderson has done is two things, for the two different sorts of customization that MCC needs.

For customization per environment, he has changed his stack to directly do the customization needed per environment.

For customization per WEC, he has switched to using mailbox WDSes. Each environment has two WDSes, each getting pre-customized workload objects AND customized a BindingPolicy (customized to match only one WEC).

In short, the customization per WEC is still needed.

**Comment by ezrasilvera** [comment]
@MikeSpreitzer You re-opened it. Can we close this and continue tracking through #1897 ?  The case described here is is a regular parameter expansion example that is covered in 1897

**Comment by ezrasilvera** [comment]
This is covered by  #1897 .  If there is an important requirement info that is not covered in #1897 , please add the missing details  as a comment into that issue. 
I went through the details here and I think the requirements in #1897 cover the needs described here. @MikeSpreitzer if you find gaps feel free to add comments to #1897
